tracer/src/Datadog.Trace/Iast/CookieAnalyzer.cs

```diff
@@ -114,26 +156,32 @@ private static void ReportVulnerabilities(IntegrationId integrationId, SetCookie
             return;
         }
 
+        bool isFiltered = IsFiltered(name);
+
         if (cookie.SameSite != Microsoft.Net.Http.Headers.SameSiteMode.Strict)
         {
-            IastModule.OnNoSamesiteCookie(integrationId, name);
+            IastModule.OnNoSamesiteCookie(integrationId, name, isFiltered);
         }
 
         if (!cookie.HttpOnly)
         {
-            IastModule.OnNoHttpOnlyCookie(integrationId, name);
+            IastModule.OnNoHttpOnlyCookie(integrationId, name, isFiltered);
         }
 
         if (!cookie.Secure)
         {
-            IastModule.OnInsecureCookie(integrationId, name);
+            IastModule.OnInsecureCookie(integrationId, name, isFiltered);
         }
     }
 
-    private static bool IsExcluded(string cookieName)
+    internal bool IsExcluded(string cookieName)
     {
         return string.IsNullOrWhiteSpace(cookieName) || cookieName.StartsWith(".AspNetCore.", StringComparison.OrdinalIgnoreCase);
     }
-
 #endif
+
+    internal bool IsFiltered(string cookieName)
+    {
+        return _cookieFilterRegex?.IsMatch(cookieName) ?? false;
```

Comment:
I think you need to catch the potential timeout here?

Reply:
Agree